### PR TITLE
File Explorer and List Files Operator

### PR DESCRIPTION
**Pull request by ritch**

### CODE REVIEW

File: app/packages/core/src/plugins/SchemaIO/components/FileExplorerView/icons.tsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
+export const AzureIcon = ({ size }) => (
```
Comment (imanjra) [inline app/packages/core/src/plugins/SchemaIO/components/FileExplorerView/icons.tsx:1]: Are these still used?

File: app/packages/operators/src/types.ts [hunk 2]
```diff
@@ -387,6 +387,35 @@ export class Trigger extends BaseType {
   }
 }
 
+/**
+ * Operator type for defining a file and its metadata.
+ */
+
+export class File extends OperatorObject {
```
Comment (imanjra) [inline app/packages/operators/src/types.ts:394]: Cool 💥

File: app/packages/plugins/package.json [hunk 3]
```diff
@@ -26,5 +26,8 @@
         "prettier": "2.2.1",
         "typescript": "4.2.4",
         "vite": "2.4.2"
+    },
+    "dependencies": {
+        "moment": "^2.29.4"
```
Comment (imanjra) [inline app/packages/plugins/package.json:31]: nit: should move to operator package

File: fiftyone/core/storage.py [hunk 4]
```diff
@@ -624,6 +624,16 @@ def list_buckets(fs, abs_paths=False):
...
 
 
+def get_available_file_systems():
+    """Returns the list of file systems that are currently available for use
+    with methods like :func:`list_files` and :func:`list_buckets`.
+
+    Returns:
+        a list of :class:`FileSystem` values
+    """
+    return [FileSystem.LOCAL]
```
Comment (imanjra) [inline fiftyone/core/storage.py:634]: 👍

File: fiftyone/operators/builtin.py [hunk 5]
```diff
@@ -284,11 +288,62 @@ def execute(self, ctx):
...
+                stat.st_mtime
+            ).isoformat(),
+            "type": "file" if os.path.isfile(filepath) else "directory",
+            "size": stat.st_size,
+            "absolute_path": os.path.join(path, filename),
+        }
+        for filename, filepath, stat in zip(filenames, filepaths, stats)
+    ]
+
+    files.sort(key=lambda file: file["type"])
```
Comment (imanjra) [inline fiftyone/operators/builtin.py:336]: 👍

**Review by imanjra — APPROVED** [review]
Left some minor comments. LGTM

File: app/packages/core/src/plugins/SchemaIO/components/FileExplorerView/ExplorerActions.tsx [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+            onPathChange(ev.target.value);
+            ev.preventDefault();
+          }
+        }}
+        variant="outlined"
+        fullWidth
+      />
+      <IconButton
+        onClick={() =>
+          onPathChange(currentPath.split("/").slice(0, -1).join("/"))
```
[the remark this answers is not in the document]
Comment (ritch) [reply app/packages/core/src/plugins/SchemaIO/components/FileExplorerView/ExplorerActions.tsx]: Done!

**Pull request merged**